SofaKernel/modules/SofaBaseMechanics/BarycentricMappers/BarycentricMapperTopologyContainer.h

```diff
@@ -90,17 +91,47 @@ class BarycentricMapperTopologyContainer : public TopologyBarycentricMapper<In,O
 
 protected:
 
+    struct HashEntry
+    {
+        HashEntry(const int& xId, const int& yId, const int& zId, const int& elementId)
+        {
+            this->xId=xId;
+            this->yId=yId;
+            this->zId=zId;
+            this->elementId=elementId;
+        }
+
+        int xId,yId,zId; // cell indices
+        int elementId;
+    };
+
+    struct NearestParams
+    {
+        NearestParams()
+        {
+            distance = std::numeric_limits<double>::max();
+            elementIndex = -1;
+        }
+
+        Vector3 baryCoords;
+        double distance;
+        int elementIndex;
+    };
+
     using Inherit1::m_fromTopology;
 
     topology::PointData< helper::vector<MappingDataType > > d_map;
     MatrixType* m_matrixJ {nullptr};
     bool m_updateJ {false};
 
+    helper::vector<Mat3x3d> m_bases;
+    helper::vector<Vector3> m_centers;
+
     // Spacial hashing utils
     Real m_gridCellSize;
     Real m_convFactor;
     unsigned int m_hashTableSize;
-    helper::vector<helper::vector<unsigned int>> m_hashTable;
+    helper::vector<helper::vector<HashEntry>> m_hashTable;
```

Comment:
Why not use `std::map` or `std::unordered_map` ? Is there a need to reimplement this ?

Reply:
std::map is not an hash-table but the remark about std::unorderd_map is valid.